Include/internal/pycore_pythread.h

```diff
@@ -106,6 +106,36 @@ PyAPI_FUNC(PyLockStatus) PyThread_acquire_lock_timed_with_retries(
     PyThread_type_lock,
     PY_TIMEOUT_T microseconds);
 
+PyAPI_FUNC(unsigned long long) PyThread_get_thread_ident_ex(void);
```

Comment:
Maybe we should typedef the return type here, so we don't have to update all the places this PR does if we change the type again later:

```suggestion
typedef unsigned long long PyThread_ident_t;

PyAPI_FUNC(PyThread_ident_t) PyThread_get_thread_ident_ex(void);
```

...and update the `PyThread_start_joinable_thread()` signature and fix the many places you changed `unsigned long` to `unsigned long long`.